test/functional/rewriteField.test.ts

```diff
@@ -0,0 +1,727 @@
+            title {
+              text
+            }
+            color
+          }
+        }
+      }
+    `;
+    expect(handler.rewriteRequest(query)).toEqual({
+      query: expectedRewritenQuery
+    });
+    expect(
+      handler.rewriteResponse({
+        theThing: {
+          thingField: {
+            id: 1,
+            title: {
+              text: 'THING'
+            },
+            color: 'blue'
+          }
+        }
+      })
+    ).toEqual({
+      theThing: {
+        thingField: {
+          id: 1,
+          title: 'THING',
+          color: 'blue'
+        }
+      }
+    });
+  });
+
+  it('rewrites a scalar field to be a renamed object field with 1 scalar subfield', () => {
+    const handler = new RewriteHandler([
+      new FieldRewriter({
+        fieldName: 'subField',
+        newFieldName: 'renamedSubField',
+        objectFieldName: 'value'
+      })
+    ]);
+
+    const query = gqlFmt`
+      query getTheThing {
+        theThing {
+          thingField {
+            id
+            subField
+            color
+          }
+        }
+      }
+    `;
+    const expectedRewritenQuery = gqlFmt`
+      query getTheThing {
+        theThing {
+          thingField {
+            id
+            renamedSubField {
+              value
+            }
+            color
+          }
+        }
+      }
+    `;
+    expect(handler.rewriteRequest(query)).toEqual({
+      query: expectedRewritenQuery
+    });
+    expect(
+      handler.rewriteResponse({
+        theThing: {
+          thingField: {
+            id: 1,
+            renamedSubField: {
+              value: 'THING'
+            },
+            color: 'blue'
+          }
+        }
+      })
+    ).toEqual({
+      theThing: {
+        thingField: {
+          id: 1,
+          subField: 'THING',
+          color: 'blue'
+        }
+      }
+    });
+  });
+
+  it('works with aliased fields', () => {
+    const handler = new RewriteHandler([
+      new FieldRewriter({
+        fieldName: 'subField',
+        newFieldName: 'renamedSubField',
+        objectFieldName: 'value'
+      })
+    ]);
+
+    const query = gqlFmt`
+      query getTheThing {
+        agg: anotheThing {
+          subField
+        }
+      }
+    `;
+    const expectedRewritenQuery = gqlFmt`
+      query getTheThing {
+        agg: anotheThing {
+          renamedSubField {
+            value
+          }
+        }
+      }
+    `;
+    expect(handler.rewriteRequest(query)).toEqual({
+      query: expectedRewritenQuery
+    });
+    expect(
+      handler.rewriteResponse({
+        agg: {
+          renamedSubField: {
+            value: 'THING'
+          }
+        }
+      })
+    ).toEqual({
+      agg: {
+        subField: 'THING'
+      }
+    });
+  });
+
+  it('works using alias for new field names', () => {
+    const handler = new RewriteHandler([
+      new FieldRewriter({
+        fieldName: 'subField',
+        newFieldName: 'aliasedField: renamedSubField',
```

Comment:
requiring the alias name in the rewriter definition like this feels clunky, since this will always rewrite the result to have the alias no matter what was sent for the old field. What's the use-case for this?

Reply:
So the alias is not required, this is optional, and detected the same way graphql does (via the semicolon character).